docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,194 @@
+         self.check_validity(self.uut,
+                             ["a = 1 + 1 + 1 + 1 + 1 + 1 + 1"],
+                             valid=False)
+
+     def test_indent_level(self):
+         test_code = ['def func():\n',
+                      '    pass\n']
+         self.check_validity(self.uut, test_code)
+
+         self.section.append(Setting('tab_width', '2'))
+         self.check_validity(self.uut, test_code, valid=False)
+         self.check_validity(self.uut, ['def func():\n', '  pass\n'])
+
+     def test_disable_warnings(self):
+         test_code = ['def func():\n',
+                      '    pass\n',
+                      'def func2():\n',
+                      '    pass\n']
+         self.check_validity(self.uut, test_code, valid=False)
+
+         self.section.append(Setting('pep_ignore', 'E302'))
+         self.check_validity(self.uut, test_code)
+
+     def test_invalid(self):
+         self.check_validity(self.uut, [""], valid=False)
+         self.check_validity(self.uut, ["a=1+1"], valid=False)
+
+First a *Section* and your Bear(in this case PEP8Bear) is setUp.Now you're
+*Section* consists on default *Settings*.You can append any *Setting* depending
+on your test.Give you're bear which you have setUp and *lines* to check with few
+additional parameters if necessary as input to check_validity.
+
+**EXAMPLE 3** using ``LocalBearTestHelper`` class(using check_results method)
+
+*LineCountBear*
+
+.. code::
+
+ from queue import Queue
+
+ from bears.general.LineCountBear import LineCountBear
+ from tests.LocalBearTestHelper import LocalBearTestHelper
+ from coalib.results.Result import RESULT_SEVERITY, Result
+ from coalib.settings.Section import Section
+
+
+ class LineCountBearTest(LocalBearTestHelper):
+
+     def setUp(self):
+         self.uut = LineCountBear(Section("name"), Queue())
+
+     def test_run(self):
+         self.check_results(
+             self.uut,
+             ["1", "2", "3"],
+             [Result.from_values("LineCountBear",
+                                 "This file has {count} lines.".format(count=3),
+                                 severity=RESULT_SEVERITY.INFO,
+                                 file="default")],
+             filename="default")
+         self.check_results(
+             self.uut,
+             [],
+             [Result.from_values("LineCountBear",
+                                 "This file has {count} lines.".format(count=0),
+                                 severity=RESULT_SEVERITY.INFO,
+                                 file="default")],
+             filename="default")
+
+Just like the above example we need to setUp a *Section* and your Bear with
+some *Settings*. ``check_results`` validates you're results, so you need to give
+your local bear, *lines* to check and expected results as input.
+
+A Final Note
+------------
+
+LocalBearTestHelper is written to ease off testing for bears.Make sure that your
+tests have a 100% coverage and zero redundancy.
```

Comment:
you should also mention that tests with `check_results` are usually better, they ensure that the *right* results are there, the other one only asserts that there are *any* or *none*